src/main/java/randoop/main/GenInputsAbstract.java

```diff
@@ -1147,6 +1173,145 @@ private static boolean shouldOmitClass(String classname) {
+    String packageNameAsFile = packageName.replace(".", File.separator) /*.concat(File.separator)*/;
+    File packageDirectory =
+        new File(directory.getPath().concat(File.separator).concat(packageNameAsFile));
+    if (packageDirectory.exists()) {
+      try (Stream<Path> allFiles = Files.walk(Paths.get(packageDirectory.getPath()))) {
+        @SuppressWarnings("signature")
+        Set<@ClassGetName String> classnames =
+            allFiles
+                .filter(e -> e.toFile().isFile()) // to ignore directories
+                .map(Path::toString)
+                .map(
+                    absolutePath ->
+                        absolutePath.replace(
+                            directory.getPath() + File.separator, "")) // to delete absolute path
+                .filter(
+                    relativePath -> relativePath.startsWith(packageNameAsFile)) // to filter package
+                .filter(relativePath -> relativePath.endsWith(".class")) // to only get classes
+                .map(
+                    relativePath ->
+                        relativePath.substring(0, relativePath.length() - ".class".length()))
+                .map(ifClassName -> ifClassName.replace(File.separator, "."))
+                .filter(
+                    classname -> {
+                      try {
+                        return classname.startsWith(packageName)
+                            && !classname
+                                .substring(packageNameAsFile.length() + 1)
+                                .contains(".") // not in subpackage
+                            && accessibility.isAccessible(Class.forName(classname));
+                      } catch (ClassNotFoundException e) {
+                        throw new RandoopClassNameError(
+                            classname,
+                            String.format(
+                                "Cannot load class found in directory %s",
+                                directory.getAbsolutePath()));
+                      }
+                    })
+                .collect(Collectors.toSet());
+        return classnames;
+      } catch (IOException e) {
+        throw new RandoopUsageError(
+            String.format(
+                "Caught an exception while reading directory %s", directory.getAbsolutePath()),
+            e);
+      }
+    }
+    return Collections.emptySet();
+  }
+
+  /**
+   * Returns classes with the given package in the given jar file.
+   *
+   * @param jarFile a jar file containing classes
+   * @param packageName a package name
+   * @param accessibility the accessibility predicate
+   * @return classes in package {@code packageName} in the given jar file
+   */
+  public static Set<@ClassGetName String> searchInJar(
+      File jarFile, String packageName, AccessibilityPredicate accessibility) {
+    Set<@ClassGetName String> classnames = new HashSet<>();
+    @ClassGetName String classname = "randoop.main.GenInputsAbstract";
```

Comment:
Is the `@ClassGetName` type annotation necessary?  I would expect it to be inferred.

Reply:
Removed type annotation, works fine.